### ISSUE TITLE
Header download phase max outs a single CPU core.

### ISSUE THREAD

**Issue by farazshaikh** [paradigmxyz/reth#1535]
### Describe the bug

Investigate CPU bottleneck in header download phase.

The header verification phase may be inherently sequential, and that stages max out a single CPU core.
i.e. Downloads are very fast but overall the performance is bottle necked by CPU. 

![image](https://user-images.githubusercontent.com/2507850/221070155-502b2da8-d2d0-4629-9ab9-d95f7c247ac0.png)

### Steps to reproduce

#target/releass/node db drop
#target/release/node 
#nmon 
 or 
#top -H

### Node logs

```text
na
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

main fdff1b0f02668e59580dc92456d349b00610b829 (

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by onbjerg** [comment]
@rkrasiuk @mattsse It should be possible to do the verification in two stages: 1. check roots (this can be done in parallel), 2. check sequentiallity (cheap, must be done serially) to utilize multiple cores better. However, I would rate this low priority since header downloads were very fast for me regardless

**Comment by rkrasiuk** [comment]
@onbjerg there are minor things we could do here for sure. just need to be cautious around deconstructing consensus abstraction, since splitting validation would require that.

**Comment by rkrasiuk** [comment]
@onbjerg which roots are you talking about btw? or did you just mean hashes?

**Comment by onbjerg** [comment]
Ah yeah, sorry, I meant hashes. But I assume this is also the case for bodies

**Comment by gakonst** [comment]
Given the headers stage is not the bottleneck I recommend we put this as low priority - Roman's fix should help a bit here
